src/js/components/Button/StyledButtonKind.js

```diff
@@ -0,0 +1,221 @@
+import styled, { css } from 'styled-components';
+
+import {
+  backgroundStyle,
+  focusStyle,
+  genericStyles,
+  normalizeColor,
+} from '../../utils';
+import { defaultProps } from '../../default-props';
+
+const radiusStyle = props => {
+  const size = props.sizeProp;
+  if (size && props.theme.button.size && props.theme.button.size[size]) {
+    return props.theme.button.size[size].border.radius;
+  }
+  return props.theme.button.border.radius;
+};
+
+const fontStyle = props => {
+  const size = props.sizeProp || 'medium';
+  const data = props.theme.text[size];
+  return css`
+    font-size: ${data.size};
+    line-height: ${data.height};
+  `;
+};
+
+const padStyle = ({ sizeProp: size, theme }) => {
+  if (size && theme.button.size && theme.button.size[size]) {
+    return css`
+      padding: ${theme.button.size[size].padding.vertical}
+        ${theme.button.size[size].padding.horizontal};
+    `;
+  }
+
+  if (theme.button.padding) {
+    return css`
+      padding: ${theme.global.edgeSize[theme.button.padding.vertical] ||
+          theme.button.padding.vertical}
+        ${theme.global.edgeSize[theme.button.padding.horizontal] ||
+          theme.button.padding.horizontal};
+    `;
+  }
+  return '';
+};
+
+// The > svg rule is to ensure Buttons with just an icon don't add additional
+// vertical height internally.
+const basicStyle = props => css`
+  border: none;
+  border-radius: ${radiusStyle(props)};
+  ${padStyle(props)}
+  ${fontStyle(props)}
+
+  > svg {
+    vertical-align: bottom;
+  }
+`;
+
+// CSS for this sub-object in the theme
+const kindPartStyles = (obj, theme, colorValue) => {
+  const styles = [];
+  if (obj.padding) {
+    if (obj.padding.vertical)
+      styles.push(
+        `padding: ${theme.global.edgeSize[obj.padding.vertical] ||
+          obj.padding.vertical ||
+          0} ${theme.global.edgeSize[obj.padding.horizontal] ||
```

Comment:
If the user defines `obj.padding.vertical` but not `obj.padding.horizontal` we are assuming the `horizontal` value is '0' rather than falling back to `button.padding.horizontal`? just confirming.

Reply:
Correct. I've updated the test such that they can specify either `horizontal` or `vertical`.